resources/external_control.urscript

```diff
@@ -95,14 +110,73 @@ thread speedThread():
   stopj(5.0)
 end
 
+thread jointTrajectoryThread():
+  textmsg("Executing trajectory")
+  textmsg(trajectory_points_left)
+  while trajectory_points_left > 0:
+    textmsg("Waiting to read point")
+    enter_critical
+    #reading trajectory point + blend radius + type of point (cartesian/joint based)
+    raw_point = socket_read_binary_integer(TRAJECTORY_DATA_DIMENSION+1+1, "trajectory_socket")
+    trajectory_points_left = trajectory_points_left - 1
+    exit_critical
+    if raw_point[0] > 0:
+      q = [ raw_point[1] / MULT_jointstate, raw_point[2] / MULT_jointstate, raw_point[3] / MULT_jointstate, raw_point[4] / MULT_jointstate, raw_point[5] / MULT_jointstate, raw_point[6] / MULT_jointstate]
+      tmptime = raw_point[7] / MULT_time
+      blend_radius = raw_point[8] / MULT_time
+
+      textmsg("Executing Point")
+      textmsg(trajectory_points_left)
+      textmsg(q)
+      textmsg(tmptime)
+      textmsg(blend_radius)
+      textmsg(raw_point[9])
+
+      if trajectory_points_left > 0:
+        if raw_point[9] == TRAJECTORY_POINT_JOINT:
+          textmsg("Executing movej with blending")
+          movej(q, t=tmptime, r=blend_radius)
+          textmsg("done")
+        elif raw_point[9] == TRAJECTORY_POINT_CARTESIAN:
+          textmsg("Executing movel with blending")
+          movel(p[q[0], q[1], q[2], q[3], q[4], q[5]], t=tmptime, r=blend_radius)
+        end
+      else:
+        if raw_point[9] == TRAJECTORY_POINT_JOINT:
+          textmsg("Executing movej without blending")
+          movej(q, t=tmptime, r=0.0)
+          textmsg("done")
+        elif raw_point[9] == TRAJECTORY_POINT_CARTESIAN:
+          textmsg("Executing movel without blending")
+          movel(p[q[0], q[1], q[2], q[3], q[4], q[5]], t=tmptime, r=0.0)
+        end
+      end
+    #add some form of error handling here
+    end
+  end
+  socket_send_int(TRAJECTORY_RESULT_SUCCESS, "trajectory_socket")
+  textmsg("Trajectory finished")
+end
+
+def clear_remaining_trajectory_points():
+  while trajectory_points_left > 0:
+    raw_point = socket_read_binary_integer(TRAJECTORY_DATA_DIMENSION+2, "trajectory_socket")
+    trajectory_points_left = trajectory_points_left - 1
+  end
+end
+
 # HEADER_END
 
 # NODE_CONTROL_LOOP_BEGINS
 
 socket_open("{{SERVER_IP_REPLACE}}", {{SERVER_PORT_REPLACE}}, "reverse_socket")
+trajectory_port = {{SERVER_PORT_REPLACE}}+10
```

Comment:
With the additional trajectory socket leaving beta, it might be worth considering if this is solution for setting the port is sufficient, or if we should add a way to explicitly set it via the setup.

Reply:
Almost forgot about that... Yes, I do not like this at all. I also don't like introducing yet another parameter, but I guess it's necessary.

Reply:
Added another parameter in ca8fc29